awx/main/analytics/core.py

```diff
@@ -116,6 +116,51 @@ def package(target, data, timestamp):
         return None
 
 
+def calculate_collection_interval(since, until):
```

Comment:
reading through this again, after spending a few days away from it, it reads well / makes a lot of sense. this feels solid 👍

Reply:
.. and it's awesome to see all this logic refactored into its own method